### PR TITLE
Fix discussion form not showing in empty discussions

### PR DESCRIPTION
**Pull request by akirk**
#### Problem

On WordPress trunk our "hack" for displaying the comment form without an existing (shadow) post no longer works (broken by https://github.com/WordPress/WordPress/commit/1069ac4afda821742cf7f600412aacc139013a55). Thus no comment form is displayed if there is no other discussion. We also want to avoid creating shadow posts just by visiting the page, so that we don't end up with many posts without discussion.

#### Solution

This PR attempts to solve this although only haphazardly so far (by checking for `$post_id === 0`).

#### Testing instructions

Go to a discussion page of an original that doesn't have a discussion yet. The comment form should be visible.

### CODE REVIEW

**Comment by akirk** [comment]
While it might not be the most elegant solution for now, it's good for testing to confirm that it really works.

File: helpers/helper-translation-discussion.php [hunk 1]
```diff
@@ -251,6 +252,13 @@ public function rewrite_original_post_type_permalink( string $post_link, WP_Post
 		return $cache[ $post->ID ];
 	}
 
+	public function comments_open_override( $open, $post_id ) {
```
Comment (amieiro) [inline helpers/helper-translation-discussion.php:255]: Can you add the header for this function?

File: includes/class-gth-temporary-post.php [hunk 2]
```diff
@@ -3,6 +3,8 @@
 class Gth_Temporary_Post {
 	public $ID;
 	public $comments_open = 'open';
+	public $comment_status = 'open';
```
Comment (amieiro) [inline includes/class-gth-temporary-post.php:6]: In a local installation wit WordPress 6.1-alpha-53873 and GlotPress 4.0.0-alpha.1, it works without these 2 new lines. Are they necessary for some edge case, or can we remove it?

**Review by amieiro — APPROVED** [review]
I have tested in a local installation wit WordPress 6.1-alpha-53873 and GlotPress 4.0.0-alpha.1, and it works fine!

File: helpers/helper-translation-discussion.php (hunk 1 above)
Comment (akirk) [reply helpers/helper-translation-discussion.php:271]: Thank you, added.

File: includes/class-gth-temporary-post.php (hunk 2 above)
Comment (akirk) [reply includes/class-gth-temporary-post.php]: I removed them, I first thought they might have been the root of the trouble.

**Comment by akirk** [comment]
@amieiro Thanks for the review. I actually had another idea of making this more robust in https://github.com/GlotPress/gp-translation-helpers/pull/106/commits/18eed41f47f9cb1d6ab7f27f5798ca9494ae5048

By just remembering that we just created a temporary post, we don't need to have comments appear for all non-existing posts but just for our case.

**Pull request merged**